src/libraries/Microsoft.Extensions.Http/src/ActiveHandlerTrackingEntry.cs

```diff
@@ -89,5 +95,44 @@ private void Timer_Tick()
                 }
             }
         }
+
+        public void StopTimer()
+        {
+            lock (_lock)
+            {
+                _timer?.Dispose();
+                _timer = null;
+            }
+        }
+
+        public void Dispose()
+        {
+            if (_disposed)
+            {
+                return;
+            }
+
+            _disposed = true;
+            StopTimer();
+
+            // When this entry is converted to an expired entry, we don't dispose anything
```

Comment:
this is currently not true, since the handlers are only nulled out here in Dispose

on the side note, i think it was a bad idea to null them out, since the active entry can be used at the same time. let's solve it another way